Fix official installer on macOS

jdpurcell opened the pull request:
Fixes #889. In addition to the incorrect download URL, the file we get is a macOS `.dmg` (disk image) which cannot be executed directly, so this handles mounting/extracting the contents. After these changes, I was able to successfully complete a [macOS build](https://github.com/jdpurcell/QtHelloWorld/actions/runs/13610178522/job/38046297825) via GitHub Actions with the official installer.

miurahr commented:
@jdpurcell  Could you also add a hash check for the downloaded binary as same as `install-qt` installer?

For example, you can get a hash value (for human) from URL like
https://download.qt.io/official_releases/online_installers/qt-unified-windows-x64-online.exe.mirrorlist

for machine processing
https://download.qt.io/official_releases/online_installers/qt-unified-mac-x64-online.dmg.sha256

miurahr commented:
The test seems failed with 

```
  =================================== FAILURES ===================================
  ______________________ test_commercial_installer_download ______________________
  
  monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f6eec4544d0>
  commercial_installer = <aqt.commercial.CommercialInstaller object at 0x7f6eec455d00>
  
      @pytest.mark.enable_socket
      def test_commercial_installer_download(monkeypatch, commercial_installer):
          """Test downloading of commercial installer"""
      
          def mock_requests_get(*args, **kwargs):
              return MockResponse(content=b"installer_content")
      
          monkeypatch.setattr(requests, "get", mock_requests_get)
      
          with TemporaryDirectory() as temp_dir:
              target_path = Path(temp_dir) / "qt-installer"
  >           commercial_installer.download_installer(target_path, timeout=60)
  E           TypeError: CommercialInstaller.download_installer() got an unexpected keyword argument 'timeout'
  
  tests/test_commercial.py:219: TypeError
  ```

jdpurcell commented:
I removed the test that failed; I don't think it's worth fixing as it wasn't really testing anything, I think its purpose was just to inflate the coverage number.

As for the hash thing, I will politely decline. I did spend a while just now trying, but it's more complicated than I can figure out in a reasonable amount of time. To be honest, I don't normally work with Python, and I'd like to keep it that way 😅.

The download code wasn't new by the way, I didn't change anything with the way downloads happen, I just moved some existing code to clean up and reduce duplication. The only new thing I added is the .dmg mounting/extraction.

jdpurcell commented:
@miurahr I rebased on `master`.
